### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by harshkhandeparkar**
No need to say thank you

### CODE REVIEW

**Comment by chirag-ghosh** [comment]
Mujhe kaun thank you bolega?

**Comment by harshkhandeparkar** [comment]
> Mujhe kaun thank you bolega?

`@everyone` thank Chirag, this was possible only because of Chirag.

File: src/pages/OAuth.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	const authContext = useAuthContext();
+	const [error, setError] = useState<string|null>(null);
+	const navigate = useNavigate();
+
+	const loginHandler = async (oauthCode: string) => {
+		// Assuming type is already set when login is started
+		const type = authContext.userData.type;
+		const auth = await makeOAuthRequest(oauthCode, type);
+
+		if ('code' in auth) {
```
Comment (chirag-ghosh) [inline src/pages/OAuth.tsx:16]: change `'code'` to something else like `'errorCode'`

Comment (harshkhandeparkar) [reply src/pages/OAuth.tsx:16]: @YoganshSharma

File: src/util/backend.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+export interface IErrorResponse {
+	code: number;
+	message: string;
+}
+
+async function makeBackendRequest(
+	endpoint: string,
+	method: 'get' | 'post',
+	jwt: string | null,
```
Comment (chirag-ghosh) [inline src/util/backend.ts:11]: Handle this internally in the function or create a wrapper which handles authenticated requests

**Review by chirag-ghosh — COMMENTED** [review]
Looks Almost Good To Me

File: src/util/backend.ts (hunk 2 above)
Comment (harshkhandeparkar) [reply src/util/backend.ts:11]: Will be handled in https://github.com/kossiitkgp/KWoC-Frontend/issues/149.

File: src/pages/OAuth.tsx (hunk 1 above)
Comment (harshkhandeparkar) [reply src/pages/OAuth.tsx:16]: Will be handled in https://github.com/kossiitkgp/KWoC-Backend/issues/160.

**Review by chirag-ghosh — APPROVED** [review]
Looks Good Enough To Me

**Pull request merged**